src/main/java/org/opensearch/knn/index/engine/faiss/FaissSQEncoder.java

```diff
@@ -30,17 +32,48 @@ public class FaissSQEncoder implements Encoder {
 
     private final static MethodComponent METHOD_COMPONENT = MethodComponent.Builder.builder(ENCODER_SQ)
         .addSupportedDataTypes(SUPPORTED_DATA_TYPES)
-        .addParameter(
-            FAISS_SQ_TYPE,
-            new Parameter.StringParameter(FAISS_SQ_TYPE, FAISS_SQ_ENCODER_FP16, (v, context) -> FAISS_SQ_ENCODER_TYPES.contains(v))
-        )
-        .addParameter(FAISS_SQ_CLIP, new Parameter.BooleanParameter(FAISS_SQ_CLIP, false, (v, context) -> Objects.nonNull(v)))
-        .setKnnLibraryIndexingContextGenerator(
-            ((methodComponent, methodComponentContext, knnMethodConfigContext) -> MethodAsMapBuilder.builder(
-                FAISS_SQ_DESCRIPTION,
+        .addParameter(FAISS_SQ_TYPE, new Parameter.StringParameter(FAISS_SQ_TYPE, (v, context) -> {
+            String vResolved = v;
+            if (vResolved == null) {
+                vResolved = FAISS_SQ_ENCODER_FP16;
+            }
+            if (FAISS_SQ_ENCODER_FP16.equals(vResolved) == false && context.getPerDimensionProcessor() == CLIP_TO_FP16_PROCESSOR) {
```

Comment:
This looks like a redundant check as you are already validating it below for clipping

Reply:
I added this because it will kind of depend in what order the parameters are parsed and figured its better to do it twice.